lib/plugin/stepByStepReport.js

```diff
@@ -135,10 +138,14 @@ module.exports = function (config) {
     if (isStepIgnored(step)) return;
     if (savedStep === step) return; // already saved
     // Ignore steps from BeforeSuite function
+    if (scenarioFailed && config.disableScreenshotOnFail) return;
```

Comment:
What would be use case for this?

It should only save screenshot for the first failure. ? if so what's the point in execution of other tests, you could just use fail fast mode.

Mocha test runner could do that with following config:
```
  mocha: {
    bail: false,
  },

```

Reply:
Was the intention to only save screenshot for the first failure? Yes it was. I never used mocha I didn't knew that, but i dont think its hurting to having this option too.